### PR TITLE
[LgTv] Trap IllegalArgumentException during deactivate()

### LINKED ISSUE #5381: [LgTv] Exception upon startup: "The deactivate method has thrown an exception"
## Expected Behavior
The binding loads and works

## Current Behavior
When I'm booting up, I get the follwing JAVA exception:
`2017-12-08 20:42:29.936 [INFO ] [binding.lgtv.internal.LgtvConnection] - Stopped LgTv Servlet 

2017-12-08 20:42:29.944 [ERROR] [org.openhab.binding.lgtv            ] - [org.openhab.binding.lgtv(207)] The deactivate method has thrown an exception
java.lang.IllegalArgumentException: Alias [/udap/api/event] was never registered
	at org.ops4j.pax.web.service.internal.HttpServiceStarted.unregister(HttpServiceStarted.java:313) ~[?:?]
	at org.ops4j.pax.web.service.internal.HttpServiceProxy.unregister(HttpServiceProxy.java:78) ~[?:?]
	at org.openhab.binding.lgtv.lginteraction.LgTvMessageReader.deactivate(LgTvMessageReader.java:178) ~[?:?]
at org.openhab.binding.lgtv.lginteraction.LgTvMessageReader.stopserver(LgTvMessageReader.java:195) ~[?:?]
at org.openhab.binding.lgtv.internal.LgtvConnection.closeConnection(LgtvConnection.java:158) ~[?:?]
at org.openhab.binding.lgtv.internal.LgtvBinding.closeAllConnections(LgtvBinding.java:378) ~[?:?]
at org.openhab.binding.lgtv.internal.LgtvBinding.deactivate(LgtvBinding.java:94) ~[?:?]
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
at org.apache.felix.scr.impl.inject.BaseMethod.invokeMethod(BaseMethod.java:229) ~[?:?]
at org.apache.felix.scr.impl.inject.BaseMethod.access$500(BaseMethod.java:39) ~[?:?]
at org.apache.felix.scr.impl.inject.BaseMethod$Resolved.invoke(BaseMethod.java:650) ~[?:?]
at org.apache.felix.scr.impl.inject.BaseMethod.invoke(BaseMethod.java:506) ~[?:?]
at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:307) ~[?:?]
at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:299) ~[?:?]
at org.apache.felix.scr.impl.manager.SingleComponentManager.disposeImplementationObject(SingleComponentManager.java:344) ~[?:?]
at org.apache.felix.scr.impl.manager.SingleComponentManager.deleteComponent(SingleComponentManager.java:153) ~[?:?]
at org.apache.felix.scr.impl.manager.AbstractComponentManager.doDeactivate(AbstractComponentManager.java:814) ~[?:?]
at org.apache.felix.scr.impl.manager.AbstractComponentManager.deactivateInternal(AbstractComponentManager.java:788) ~[?:?]
at org.apache.felix.scr.impl.manager.AbstractComponentManager.dispose(AbstractComponentManager.java:580) ~[?:?]
at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.disposeComponents(ConfigurableComponentHolder.java:706) ~[?:?]
at org.apache.felix.scr.impl.BundleComponentActivator.dispose(BundleComponentActivator.java:523) ~[?:?]
at org.apache.felix.scr.impl.Activator.disposeComponents(Activator.java:439) ~[?:?]
at org.apache.felix.scr.impl.Activator.access$300(Activator.java:54) ~[?:?]
at org.apache.felix.scr.impl.Activator$ScrExtension.destroy(Activator.java:293) ~[?:?]
at org.apache.felix.utils.extender.AbstractExtender$2.run(AbstractExtender.java:285) ~[?:?]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:?]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:?]
at org.apache.felix.utils.extender.AbstractExtender.destroyExtension(AbstractExtender.java:307) ~[?:?]
at org.apache.felix.utils.extender.AbstractExtender.bundleChanged(AbstractExtender.java:181) ~[?:?]
at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:903) ~[?:?]
at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) ~[?:?]
at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) ~[?:?]
at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:213) ~[?:?]
at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:120) ~[?:?]
at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:112) ~[?:?]
at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:156) ~[?:?]
at org.eclipse.osgi.container.Module.publishEvent(Module.java:476) ~[?:?]
at org.eclipse.osgi.container.Module.doStop(Module.java:634) ~[?:?]
at org.eclipse.osgi.container.Module.stop(Module.java:498) ~[?:?]
at org.eclipse.osgi.internal.framework.EquinoxBundle.stop(EquinoxBundle.java:410) ~[?:?]
at org.apache.karaf.features.internal.service.FeaturesServiceImpl.stopBundle(FeaturesServiceImpl.java:1367) ~[?:?]
at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:642) ~[?:?]
at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1248) ~[?:?]
at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$1(FeaturesServiceImpl.java:1147) ~[?:?]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
at java.lang.Thread.run(Thread.java:748) [?:?]
2017-12-08 20:42:31.040 [INFO ] [core.karaf.internal.FeatureInstaller] - Uninstalled 'openhab-binding-lgtv1'`

## Possible Solution
None. OH2 deinstalls the binding automatically after the error

## Steps to Reproduce (for bugs)
1. Just boot & configure the LG-TV binding corretly

## Context
Nothing special

## Your Environment
openHAB-Version:  openHAB 2.2.0~20171208181417-1 (Build #1119) [openHABian], expierenced the same problems unter 2.1.0
Hardware: Raspberry PI 2

Here some maybe helpful files:

**lgtv.cfg:**

> lgtvwz.host=192.168.1.xx
> lgtvwz.port=8080
> lgtvwz.pairkey=123456
> lgtvwz.serverport=8080
> lgtvwz.xmldatafiles=./
> lgtvwz.checkalive=60
> 

**items definition:**
> String LGTVWZ_AppStart          "Start"                          {lgtv="\*:lgtvwz:APP_EXECUTE"}
> String LGTVWZ_AppStopp          "Stopp"                          {lgtv="\*:lgtvwz:APP_TERMINATE"}
> String LGTVWZ_Status            "Status [%s]"                    {lgtv="\*:lgtvwz:CONNECTION_STATUS"}

**sitemap:**

> Frame label="Tv"
>     {
>         Switch item=LGTVWZ_Status
>         Switch item=LGTVWZ_AppStart mappings=[YouTube=YouTube, LOVEFiLM=Prime]
>         Switch item=LGTVWZ_AppStopp mappings=[YouTube=YouTube, LOVEFiLM=Prime]
>     }
> 

Thanks, Boby

### PR DESCRIPTION
**Pull request by 9037568**
Also cleaned up and improved logging.

Fixes #5381.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public LgTvMessageReader(int portno) {
     protected HttpService httpService;
 
     public void setHttpService(HttpService httpService) {
-        logger.info("sethttpservice called");
+        logger.info("setHttpService called");
```
Comment (kaikreuzer) [inline bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java]: could we directly reduce this to debug level?

File: bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java [hunk 2]
```diff
@@ -140,6 +140,8 @@ public String getServletInfo() {
     }
 
     public void activate() {
+        logger.info("LgTv servlet activate called");
```
Comment (kaikreuzer) [inline bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java]: also debug

File: bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java [hunk 3]
```diff
@@ -148,39 +150,42 @@ public void activate() {
...
         }
     }
 
     public void deactivate() {
-        httpService.unregister(WEBAPP_ALIAS + SERVLET_NAME);
-        logger.info("Stopped LgTv Servlet ");
+        try {
+            httpService.unregister(WEBAPP_ALIAS + SERVLET_NAME);
+        } catch(IllegalArgumentException e) {
+            logger.info("LgTv Servlet '{}' was not registered. Nothing to deactivate.", WEBAPP_ALIAS + SERVLET_NAME);
```
Comment (kaikreuzer) [inline bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java]: debug

File: bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java (hunk 1 above)
Comment (9037568) [reply bundles/binding/org.openhab.binding.lgtv/src/main/java/org/openhab/binding/lgtv/lginteraction/LgTvMessageReader.java]: No objections. Updated all of these.

**Comment by 9037568** [comment]
Thanks for the correction on the exception ;-)

**Pull request merged**